server/models/dataset_helpers/image_metadata.py

```diff
@@ -737,15 +741,31 @@ def _populateMetadata(acquisition, clinical):
 
     # Set melanocytic field based on diagnosis
     if diagnosis in MelanocyticFieldParser.melanocyticDiagnoses:
-        if clinical.get('melanocytic') is None:
+        melanocytic = clinical.get('melanocytic')
+        if melanocytic is None:
+            clinical['melanocytic'] = True
+        elif not melanocytic:
             clinical['melanocytic'] = True
+            warnings.append('corrected inconsistent value for field %r based on field %r '
+                            '(new value: %r, %r: %r)' %
+                            (MelanocyticFieldParser.name, DiagnosisFieldParser.name,
+                             True, DiagnosisFieldParser.name, diagnosis))
     elif diagnosis is not None and diagnosis != 'other':
-        if clinical.get('melanocytic') is None:
+        melanocytic = clinical.get('melanocytic')
+        if melanocytic is None:
             clinical['melanocytic'] = False
+        elif melanocytic:
+            clinical['melanocytic'] = False
+            warnings.append('corrected inconsistent value for field %r based on field %r '
+                            '(new value: %r, %r: %r)' %
+                            (MelanocyticFieldParser.name, DiagnosisFieldParser.name,
+                             True, DiagnosisFieldParser.name, diagnosis))
```

Comment:
The new value logged should probably be `False`.

Reply:
Fixed.